src/lm_robust_helper.cpp

```diff
@@ -146,24 +146,24 @@ List lm_robust_helper(const arma::vec & y,
 
       if (type == "CR2") {
 
-        // if weights
+        // Instead of having X and Xt be weighted by sqrt(W), this has them weighted by W
         Xt = Xt.each_row() % arma::sqrt(arma::trans(weights));
+        X = X.each_col() % arma::sqrt(weights);
```

Comment:
If we never use weights not sqrt'd, we might as well pass in sqrt(w) from R

Reply:
Great, there was also a duplication of effort in R. Thanks.